integration-tests/testutils/autopkgtest/autopkgtest.go

```diff
@@ -48,15 +48,18 @@ type Autopkgtest struct {
 	testArtifactsPath   string // location of the test artifacts on the host
 	testFilter          string
 	integrationTestName string
+	shellOnFail         bool
 }
 
 // NewAutopkgtest is the Autopkgtest constructor
-func NewAutopkgtest(sourceCodePath, testArtifactsPath, testFilter, integrationTestName string) *Autopkgtest {
+func NewAutopkgtest(sourceCodePath, testArtifactsPath, testFilter, integrationTestName string, shellOnFail bool) *Autopkgtest {
```

Comment:
Feels like this function isn't really helping much. The struct-building logic inside it is so much more clear than the long list of arguments that the function takes in the call sites.

Also, on the thread of general Go-related suggestions, Autopkgtest would conventionally be named AutoPkgTest.